src/components/Settings/index.tsx

```diff
@@ -51,6 +52,19 @@ export default function SettingsTab({ autoSlippage, chainId }: { autoSlippage: P
   const toggleMenu = useToggleSettingsMenu()
   useOnClickOutside(node, isOpen ? toggleMenu : undefined)
 
+  useEffect(() => {
```

Comment:
This is the same effect across both files - can you consolidate into a single hook, `useDisableScrolling(disabled: boolean)`, in `src/hooks/useDisableScrolling.ts`?